agent/consul/connect_ca_endpoint.go

```diff
@@ -177,6 +178,7 @@ func (s *ConnectCA) ConfigurationSet(
 		newRoot := *r
 		if newRoot.Active {
 			newRoot.Active = false
+			newRoot.RotateOutAt = time.Now().Add(caRootExpireDuration)
```

Comment:
I feel like it would make more sense to record the time that the CA root stopped being _active_ and then apply the delta when pruning.

This seems like a more useful but of state to record - for example when listing CA roots and showing state of a rotation it seems more logical to be able to show the time the CA stopped being actively used than have to reverse that. Especially once the delta stops being a static constant.

Are there benefits to doing it this way around I'm not seeing? If so because we want to make it explicit at time the root changes not dependent on later config changes then we could consider recording both times but it seems if there is only one then the actual time of the transition away from being active is more logical?

Reply:
I did it this way so it wouldn't be affected by later config changes. We could add the second field (the time it became inactive at) - I only left that out since it's not being used for anything yet. If we just want to be able to show that in the API though for clarity, that's probably a good enough reason to add it in.

Reply:
Yeah I guess if config changed the certificate lifetime (later when supported) to be shorter you wouldn't want this to change. That said it's already an edge cases that is not fixed completely by storing the `RotateOutAt`. Doing it this way assumes that the certificate lifetime was the same for the whole of the period for which the CA was active which is not gauranteed since the config might have been reloaded several times.

So even with it like it is, it doesn't make a strong guarantee that changing certificate lifetime will keep this safe, so I'd vote for just storing the time it stopped being active and then either leave it and document the behaviour for now, or store in the CA state on every config reload the _maximum_ cert lifetime seen during the active life of that CA?